packages/ui-components/src/lib/components/deployment/DeploymentSteps.svelte

```diff
@@ -186,6 +188,15 @@
 
 	async function _handleUpdateGuiState(gui: DotrainOrderGui) {
 		await areAllTokensSelected();
+
+		if (allTokensSelected) {
+			let newAllTokenInfos = await gui.getAllTokenInfos();
+			if (newAllTokenInfos !== allTokenInfos) {
```

Comment:
hm this isn't a deep comparison of equality, does it actually work? this is something we should show in a test

also, we are calling getAllTokenInfos with every state change, but we only need to do that if the tokens have changed

so we need more granular control over our state change callback

Reply:
I think this will be easier to implement after merging #1361. Let's visit this again after that PR